speechbrain/decoders/scorer.py

```diff
@@ -192,54 +190,46 @@ def reset_mem(self, x, enc_lens):
         return None
 
 
-class NGramLMScorer(BaseScorerInterface):
-    """A ngram LM scorer. Please make sure the tokenizer or token_list
-    matches the tokenizer of the ASR model.
+class KenLMScorer(BaseScorerInterface):
+    """KenLM N-gram scorer
 
     Arguments
     ---------
     lm_path : str
         The path of ngram model.
     vocab_size: int
         The total number of tokens.
-    tokenizer_path : str
-        The path of pre-trained sentencepiece tokenizer.
     token_list : list
         The tokens set.
     """
 
     def __init__(self, lm_path, vocab_size, tokenizer_path=None, token_list=[]):
-        self.lm = kenlm.Model(lm_path)
+        try:
+            import kenlm
+
+            self.kenlm = kenlm
+        except ImportError:
+            MSG = """Couldn't import KenLM
+            It is an optional dependency; it is not installed with SpeechBrain
+            by default. Install it with:
+            > pip install https://github.com/kpu/kenlm/archive/master.zip
+            """
+            raise ImportError(MSG)
+        self.lm = self.kenlm.Model(lm_path)
         self.vocab_size = vocab_size
         self.full_candidates = np.arange(self.vocab_size)
         self.minus_inf = -1e20
-
-        if len(token_list) > 0:
-            assert (
-                len(token_list) == vocab_size
-            ), "The size of the token_list and vocab_size are not matched."
-            self.id2char = token_list
-
-        elif tokenizer_path is not None:
-            # Create token list
-            tokenizer = spm.SentencePieceProcessor()
-            tokenizer.load(tokenizer_path)
-            self.id2char = [
-                tokenizer.id_to_piece([i])[0].replace("\u2581", "_")
-                for i in range(vocab_size)
-            ]
-
-        else:
-            raise ValueError(
-                "Please specify the token list or the path to the spm tokenizer."
-            )
+        if len(token_list) != vocab_size:
+            MSG = "The size of the token_list and vocab_size are not matched."
+            raise ValueError(MSG)
```

Comment:
Should we also handle other error messages in this way? I think this part should be consistent. Many error messages are not in this format.

Reply:
Here I changed an assert statement, and yes, mostly we should use this if-raise pattern. Asserts are meant for [debugging](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement), like an internal consistency check. In other words, if an assertion error occurs, it should mean that there is a bug in SpeechBrain code, not an error in user code.

Reply:
I will adapt this rule to other parts.